opt_adapt/opt.py

```diff
@@ -52,11 +53,33 @@ def __init__(self, options={}):
                 raise ValueError(f"Option {key} not recognised")
             self.__setattr__(key, value)
 
+# Only works for problem with control paramter in r-space
+def line_search(forward_run, mesh, u, P, J, dJ, Rspace, alpha=1e-4, max_search_iter=100):
```

Comment:
Please add a docstring that describes what the function does. The comment can be turned into a "to do". The way I would do it is to just change it to
```
# TODO: Only works for problems with control parameters in R-space
```